File: backend/src/main/java/com/gxs/bank/service/KycService.java
package com.gxs.bank.service;

import com.gxs.bank.exception.ResourceNotFoundException;
import com.gxs.bank.model.KycDocument;
import com.gxs.bank.model.User;
import com.gxs.bank.repository.KycDocumentRepository;
import com.gxs.bank.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class KycService {
    private final KycDocumentRepository kycDocumentRepository;
    private final UserRepository userRepository;
    private final NotificationService notificationService;

    public KycDocument submitDocument(UUID userId, KycDocument.DocumentType type, String documentNumber, String fileUrl) {
        User user = userRepository.findById(userId).orElseThrow(() -> new ResourceNotFoundException("User not found"));
        
        KycDocument doc = KycDocument.builder()
            .user(user)
            .documentType(type)
            .documentNumber(documentNumber)
            .fileUrl(fileUrl)
            .verificationStatus(KycDocument.VerificationStatus.UPLOADED)
            .build();
            
        KycDocument saved = kycDocumentRepository.save(doc);
        
        user.setKycStatus(User.KycStatus.PENDING_REVIEW);
        userRepository.save(user);
        
        notificationService.createNotification(user, "KYC Submitted", "Your " + type + " has been submitted for review.", com.gxs.bank.model.Notification.Type.KYC);
        
        return saved;
    }

    public List<KycDocument> getUserDocuments(UUID userId) {
        return kycDocumentRepository.findByUserId(userId);
    }
}

File: backend-python/com/gxs/bank/service/KycService.py
from com.gxs.bank.exception.ResourceNotFoundException import ResourceNotFoundException
from com.gxs.bank.model.KycDocument import KycDocument
from com.gxs.bank.model.Notification import Notification
from com.gxs.bank.model.User import User
from com.gxs.bank.repository.KycDocumentRepository import KycDocumentRepository
from com.gxs.bank.repository.UserRepository import UserRepository
from com.gxs.bank.service.NotificationService import NotificationService


class KycService:
    def __init__(self, db):
        self.db = db
        self.kycDocumentRepository = KycDocumentRepository(db)
        self.userRepository = UserRepository(db)
        self.notificationService = NotificationService(db)

    def submitDocument(self, userId: str, documentType, documentNumber: str, fileUrl: str):
        user = self.userRepository.findById(userId)
        if user is None:
            raise ResourceNotFoundException("User not found")

        doc = KycDocument(
            user=user,
            userId=user.id,
            documentType=documentType,
            documentNumber=documentNumber,
            fileUrl=fileUrl,
            verificationStatus=KycDocument.VerificationStatus.UPLOADED,
        )
        self.kycDocumentRepository.save(doc)

        user.kycStatus = User.KycStatus.PENDING_REVIEW
        self.notificationService.createNotification(
            user,
            "KYC Submitted",
            f"Your {documentType.value} has been submitted for review.",
            Notification.Type.KYC,
            auto_commit=False,
        )
        self.db.commit()
        self.db.refresh(doc)
        return doc

    def getUserDocuments(self, userId: str):
        return self.kycDocumentRepository.findByUserId(userId)
